### PR TITLE
Added strict mode for MAGICCBase

### PR DESCRIPTION
**Pull request by lewisjared**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Description in ``CHANGELOG.rst`` added

Optional strict parameter to allow runs to continue with invalid configuration

### CODE REVIEW

File: pymagicc/core.py [hunk 1]
```diff
@@ -94,10 +94,15 @@ def __init__(self, root_dir=None):
             Root directory of the MAGICC package. If ``None``, a temporary
             copy of MAGICC is made based on the result of `
             `self.get_exectuable()``.
+        strict: bool
+            If True, enforce the configuration checks, otherwise a warning
+            is raised if any invalid configuration is found and the run is
+            continued.
```
Comment (znicholls) [inline pymagicc/core.py]:
```suggestion
            continued. Setting ``strict=False`` is only recommended for experienced users of MAGICC.
```

Comment (znicholls) [reply pymagicc/core.py]: One thought I had

**Comment by rgieseke** [comment]
What's the usecase? Legacy files?

**Comment by znicholls** [comment]
Reproducing other work flows. Best practice is to not do this, but if
you’re just trying to reproduce someone else’s work then it saves a bunch
of mucking around trying to work out how to make the two workflows match
(which is what you have to do if you don’t have the not strict option).

File: tests/test_core.py [hunk 2]
```diff
@@ -158,6 +158,75 @@ def test_run_failure_confusing_emissions_scenarios(package, invalid_config):
...
+    ],
+)
+def test_check_config_non_strict(package_non_strict, invalid_config):
+    f90nml.write(invalid_config, join(package_non_strict.run_dir, "MAGCFG_USER.CFG"), force=True)
+
+    error_msg = re.escape(
+        "You have more than one `FILE_EMISSCEN_X` flag set. Using more than one "
+        "emissions scenario is hard to debug and unnecessary with Pymagicc's "
+        "dataframe scenario input. Please combine all your scenarios into one "
+        "dataframe with Pymagicc and pandas, then feed this single Dataframe into "
```
Comment (rgieseke) [inline tests/test_core.py]:
```suggestion
        "DataFrame with Pymagicc and Pandas, then feed this single DataFrame into "
```

File: tests/test_core.py [hunk 3]
```diff
@@ -158,6 +158,75 @@ def test_run_failure_confusing_emissions_scenarios(package, invalid_config):
...
+        },
+    ],
+)
+def test_check_config_non_strict(package_non_strict, invalid_config):
+    f90nml.write(invalid_config, join(package_non_strict.run_dir, "MAGCFG_USER.CFG"), force=True)
+
+    error_msg = re.escape(
+        "You have more than one `FILE_EMISSCEN_X` flag set. Using more than one "
+        "emissions scenario is hard to debug and unnecessary with Pymagicc's "
+        "dataframe scenario input. Please combine all your scenarios into one "
```
Comment (rgieseke) [inline tests/test_core.py]:
```suggestion
        "DataFrame scenario input. Please combine all your scenarios into one "
```

**Comment by znicholls** [comment]
@lewisjared if you address the suggestions and get the tests passing I'll merge

**Comment by lewisjared** [comment]
Implemented the suggested changes. The failing tests are from the notebooks and the pyam integration which is not from this PR

**Comment by znicholls** [comment]
> The failing tests are from the notebooks and the pyam integration which is not from this PR

Can you hotfix them by just pinning pandas==0.24.2 in this PR?

**Pull request merged**